charms/jupyter-controller/src/charm.py

```diff
@@ -59,110 +80,181 @@ def __init__(self, *args):
-                                },
-                                {"apiGroups": [""], "resources": ["services"], "verbs": ["*"]},
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": [
-                                        "notebooks",
-                                        "notebooks/finalizers",
-                                        "notebooks/status",
-                                    ],
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": ["networking.istio.io"],
-                                    "resources": ["virtualservices"],
-                                    "verbs": ["*"],
-                                },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "jupyter-controller",
-                        "imageDetails": image_details,
-                        "command": ["./manager"],
-                        "envConfig": {
-                            "USE_ISTIO": "true",
-                            "ISTIO_GATEWAY": f"{model}/kubeflow-gateway",
-                            "ENABLE_CULLING": config["enable-culling"],
-                        },
-                    }
-                ],
+        ret_env_vars = {
+            "USE_ISTIO": "true",
+            "ISTIO_GATEWAY": f"{self.model.name}/kubeflow-gateway",
+            "ENABLE_CULLING": config["enable-culling"],
+        }
+
+        return ret_env_vars
+
+    @property
+    def _jupyter_controller_layer(self) -> Layer:
+        """Create and return Pebble framework layer."""
+        env_vars = self.service_environment
+
+        layer_config = {
+            "summary": "jupyter-controller layer",
+            "description": "Pebble config layer for jupyter-controller",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Entrypoint of jupyter-controller image",
+                    "command": self._exec_command,
+                    "startup": "enabled",
+                    "environment": env_vars,
+                }
             },
-            k8s_resources={
-                "kubernetesResources": {
-                    "customResourceDefinitions": [
-                        {"name": crd["metadata"]["name"], "spec": crd["spec"]}
-                        for crd in yaml.safe_load_all(Path("src/crds.yaml").read_text())
-                    ],
+            "checks": {
+                "jupyter-controller-get": {
+                    "override": "replace",
+                    "period": "30s",
+                    "http": {"url": f"http://localhost:{METRICS_PORT}{METRICS_PATH}"},
```

Comment:
Is this a health check or metric collection?  If health check, should it pull from the [probe_addr](https://github.com/kubeflow/kubeflow/blob/master/components/notebook-controller/main.go#L65) (8081) instead of the metrics_addr (8080)?

Reply:
Good point. I will take a look. In other charms we usually use either UI or metric endpoints. If there is specific one for probe, may be it makes sense to use it.